Composer/packages/lib/shared/src/types/sdk.ts

```diff
@@ -229,6 +240,32 @@ export interface SwitchCondition extends BaseSchema {
   default?: MicrosoftIDialog[];
 }
 
+/** Two-way branch the conversation flow based on a condition. */
+export interface IfCondition extends BaseSchema {
+  /** Expression to evaluate. */
+  condition?: string;
+  actions?: MicrosoftIDialog[];
+  elseActions?: MicrosoftIDialog[];
+}
+
+/** Execute actions on each item in an a collection. */
+export interface Foreach extends BaseSchema {
+  itemsProperty?: string;
+  actions?: MicrosoftIDialog[];
+}
+
+/** Execute actions on each page (collection of items) in an array. */
+export interface ForeachPage extends BaseSchema {
+  itemsProperty?: string;
+  pageSize?: number;
+  actions?: MicrosoftIDialog[];
+}
+
+export interface EditActions extends BaseSchema {
+  changeType: string;
+  actions?: MicrosoftIDialog[];
+}
+
```

Comment:
Thanks for improving these!

They should be added to the `MicrosoftIDiloag` union type as well.

Reply:
they are added, mark as resolved